packages/public/server/src/module/Taxonomy/src/Entity/TaxonomyTerm.php

```diff
@@ -320,11 +320,17 @@ public function isAssociated(TaxonomyTermAwareInterface $object)
     public function knowsAncestor(TaxonomyTermInterface $ancestor)
     {
         $term = $this;
-        while ($term->hasParent()) {
-            $term = $term->getParent();
+
+        while (true) {
             if ($term === $ancestor) {
                 return true;
             }
+
+            if ($term->hasParent()) {
+                $term = $term->getParent();
+            } else {
+                break;
+            }
```

Comment:
Why is this needed? If I understand the change correctly, this now also covers the case where `$this` is already `$ancestor` (instead of any parent). Is this a change that is conform to its usage? If so, should we maybe clarify the name? Not sure if I'd call me my own ancestor 😅

Reply:
Yes, this is the new use case -> I have renamed the method...